src/poprox_storage/repositories/datasets.py

```diff
@@ -68,46 +68,21 @@ def fetch_account_aliases(self, dataset_id: UUID) -> dict[UUID, UUID]:
         rows = self.conn.execute(query).fetchall()
         return {row.account_id: row.alias_id for row in rows}
 
-    def fetch_newsletters(self, dataset_id: UUID, start: datetime, end: datetime) -> list[Newsletter]:
-        newsletters_table = self.tables["newsletters"]
+    def fetch_newsletter_impressions(self, dataset_id: UUID, start: datetime, end: datetime) -> list[Impression]:
```

Comment:
do we want these parameters? Or does it make sense to have the newsletter_impressions fetch based on a list of newsletter ids?